biorxiv/corpora_comparison/05_figure_generator_reviewer_request.py

```diff
@@ -0,0 +1,957 @@
+)
+g.save("output/svg_files/biorxiv_pmc_frequency_odds.svg")
+g.save("output/svg_files/biorxiv_pmc_frequency_odds.png", dpi=75)
+print(g)
+
+count_plot_df = (
+    create_lemma_count_df(plot_df, "bioRxiv", "pmc")
+    .replace({"pmc": "PMC"})
+    .assign(
+        repository=lambda x: pd.Categorical(
+            x.repository.tolist(), categories=["bioRxiv", "PMC"]
+        )
+    )
+)
+count_plot_df.head()
+
+g = plot_bargraph(count_plot_df, plot_df)
+g.save("output/svg_files/biorxiv_pmc_frequency_bar.svg")
+print(g)
+
+# +
+fig_output_path = "output/figures/biorxiv_vs_pubmed_central.png"
+
+fig = plot_point_bar_figure(
+    "output/svg_files/biorxiv_pmc_frequency_odds.svg",
+    "output/svg_files/biorxiv_pmc_frequency_bar.svg",
+)
+
+# save generated SVG files
+svg2png(bytestring=fig.to_str(), write_to=fig_output_path, dpi=75)
+
+Image(fig_output_path)
+# -
+
+# ### Special Char Removed
+
+full_plot_special_char_df = calculate_confidence_intervals(
+    full_text_comparison_special_char
+)
+full_plot_special_char_df.head()
+
+plot_special_char_df = (
+    full_plot_special_char_df.sort_values("odds_ratio", ascending=False)
+    .head(subset)
+    .append(
+        full_plot_special_char_df.sort_values("odds_ratio", ascending=False).tail(
+            subset
+        )
+    )
+    .replace("rna", "RNA")
+    .assign(
+        odds_ratio=lambda x: x.odds_ratio.apply(lambda x: np.log2(x)),
+        lower_odds=lambda x: x.lower_odds.apply(lambda x: np.log2(x)),
+        upper_odds=lambda x: x.upper_odds.apply(lambda x: np.log2(x)),
+    )
+)
+plot_special_char_df.head()
+
+g = (
+    p9.ggplot(
+        plot_special_char_df,
+        p9.aes(y="lemma", x="lower_odds", xend="upper_odds", yend="lemma"),
+    )
+    + p9.geom_segment(color="#253494", size=3.5, alpha=0.7)
+    + p9.scale_y_discrete(
+        limits=(
+            plot_special_char_df.sort_values(
+                "odds_ratio", ascending=True
+            ).lemma.tolist()
+        )
+    )
+    + p9.scale_x_continuous(limits=(-3, 3))
+    + p9.geom_vline(p9.aes(xintercept=0), linetype="--", color="grey")
+    + p9.annotate(
+        "segment",
+        x=0.5,
+        xend=2.5,
+        y=1.5,
+        yend=1.5,
+        colour="black",
+        size=0.5,
+        alpha=1,
+        arrow=p9.arrow(length=0.1),
+    )
+    + p9.annotate("text", label="bioRxiv Enriched", x=1.5, y=2.5, size=18, alpha=0.7)
+    + p9.annotate(
+        "segment",
+        x=-0.5,
+        xend=-2.5,
+        y=39.5,
+        yend=39.5,
+        colour="black",
+        size=0.5,
+        alpha=1,
+        arrow=p9.arrow(length=0.1),
+    )
+    + p9.annotate("text", label="PMC Enriched", x=-1.5, y=38.5, size=18, alpha=0.7)
+    + p9.theme_seaborn(context="paper", style="ticks", font_scale=1.8, font="Arial")
+    + p9.theme(
+        figure_size=(11, 8.5),
+        panel_grid_minor=p9.element_blank(),
+    )
+    + p9.labs(y=None, x="bioRxiv vs PMC log2(Odds Ratio)")
```

Comment:
The bars for et, al, and genes seem to be missing from this plot. Maybe because their error bar is too thin?

Also, is there a difference in how you do stemming or tokenization between the analyses with and without special characters? It seems weird that one has both "patient" and "patients" in its top pubmed terms while the other only has "patients"

Reply:
>The bars for et, al, and genes seem to be missing from this plot. Maybe because their error bar is too thin?

Good question. Turns out plotnine likes to remove bars if the y-axis ticks become too large. Have updated that fix that issue.

>Also, is there a difference in how you do stemming or tokenization between the analyses with and without special characters? It seems weird that one has both "patient" and "patients" in its top pubmed terms while the other only has "patients"

Normally there is no difference when removing special characters. When I performed this analysis I didn't  have spacy lemmatize each token. Therefore, "patient" and "patients" are considered separate tokens and have equal chance of showing up for this supplemental part.